src/librustdoc/html/render/mod.rs

```diff
@@ -2410,3 +2424,198 @@ fn collect_paths_for_type(first_ty: clean::Type, cache: &Cache) -> Vec<String> {
     }
     out
 }
+
+const MAX_FULL_EXAMPLES: usize = 5;
+
+/// Generates the HTML for example call locations generated via the --scrape-examples flag.
+fn render_call_locations(
+    w: &mut Buffer,
+    cx: &Context<'_>,
+    call_locations: &Option<FnCallLocations>,
+    item: &clean::Item,
+) {
+    let call_locations = match call_locations.as_ref() {
+        Some(call_locations) if call_locations.len() > 0 => call_locations,
+        _ => {
+            return;
+        }
+    };
+
+    // Generate a unique ID so users can link to this section for a given method
+    let id = cx.id_map.borrow_mut().derive("scraped-examples");
```

Comment:
I think the "correct" way to do this for statically-generated IDs is to stick them in `init_id_map()` and just use the `scrape-examples` ID without `derive()`. @GuillaumeGomez is that right?

Reply:
That only affects perf, right? The behavior is the same?

@willcrichton if you could leave a FIXME here that would be great but I don't think it needs to be fixed immediately.

Reply:
Done!